data/src/main/java/com/microsoft/azure/kusto/data/auth/TokenProviderBase.java

```diff
@@ -16,5 +19,5 @@ public TokenProviderBase(@NotNull String clusterUrl) throws URISyntaxException {
         this.clusterUrl = UriUtils.setPathForUri(clusterUrl, "");
     }
 
-    public abstract String acquireAccessToken() throws DataServiceException, DataClientException;
+    public abstract String acquireAccessToken(@Nullable HttpClient httpClient) throws DataServiceException, DataClientException;
```

Comment:
Per our conversation, consider moving HttpClient to the C'tor.

Reply:
Looks good?